pkg/controller/petset/fakes.go

```diff
@@ -246,7 +246,7 @@ func (f *fakePetClient) deletePetAtIndex(index int) {
 }
 
 func (f *fakePetClient) setHealthy(index int) error {
-	if len(f.pets) < index {
+	if len(f.pets) <= index {
```

Comment:
sorry about that mate, I encountered it while testing my stuff and decided to fix it as well. I hope your PR gets in regardless!